### PR TITLE
Lb issue 16 xtext maven plugin its

### LINKED ISSUE #16: Fix xtext-maven-plugin ITs
While experimenting with a single Maven build, https://github.com/xtext/xtext-monorepo/issues/14 The `XtextGeneratorIT` started to show failures on Jenkins (not on GitHub Actions). The failure doesn't make sense because in the first build I didn't change anything critical.

```
21:50:51  [ERROR] Errors: 
21:50:51  [ERROR]   XtextGeneratorIT.mavenConfiguration:137->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [ERROR]   XtextGeneratorIT.simpleLang:88->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [ERROR]   XtextGeneratorIT.simpleLangIncremental:95->verifyErrorFreeLog:238->verifyErrorFreeLog:251 » Verification
21:50:51  [INFO] 
21:50:51  [ERROR] Tests run: 14, Failures: 0, Errors: 3, Skipped: 0
```

In the log, I see

```
21:50:51  [INFO] -----------------< org.eclipse.xtext:simple-lang-test >-----------------
21:50:51  [INFO] Building simple-lang-test IT-SNAPSHOT
21:50:51  [INFO] --------------------------------[ jar ]---------------------------------
21:50:51  [WARNING] The metadata /home/jenkins/.m2/repository/org/eclipse/xtext/org.eclipse.xtext.maven.parent/2.31.0-SNAPSHOT/maven-metadata-local.xml is invalid: Snapshot information corrupted with remote repository data, please verify that no remote repository uses the id 'local'
```

This message comes from https://github.com/apache/maven/blob/master/maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/DefaultVersionResolver.java#L266 and the comment says

```
 NOTE: Users occasionally misuse the id "local" for remote repos which screws up the metadata
 of the local repository. This is especially troublesome during snapshot resolution so we try
 to handle that gracefully.
```

In fact, I think we're doing the wrong thing in those integration tests.

I forgot to mention that I had already fixed problems with integration tests of the other maven plugin, xtend-maven-plugin: https://github.com/xtext/xtext-monorepo/commit/aadb4482897d943334952497d0a6bbda11a8e748

The problem is that we run the integration tests of the test projects against the same Maven local repository we use in the current build, and I guess we corrupt our local maven repository with wrong artifacts.

From what I know, the standard way to run integration tests for maven plugins is to first use the https://maven.apache.org/plugins/maven-invoker-plugin/install-mojo.html to install all the dependencies in a segregated and isolated maven repository and run the integration tests against that repository. Of course, the verifier must be set to use that local maven repository.
That's how I fixed xtend-maven-plugin https://github.com/xtext/xtext-monorepo/commit/aadb4482897d943334952497d0a6bbda11a8e748

I'll try to apply the same fix to xtext-maven-plugin

@cdietrich @szarnekow do you happen to know why we were running integration tests like that?

### PR DESCRIPTION
**Pull request by LorenzoBettini**
Closes #16

### CODE REVIEW

File: org.eclipse.xtext.maven.plugin/README.txt [hunk 1]
```diff
@@ -0,0 +1,56 @@
+To run the integration tests, the `target/.m2test` folder must first be populated with Maven dependencies and with the IT-SNAPSHOT jar of this Maven plugin.
```
Comment (cdietrich) [inline org.eclipse.xtext.maven.plugin/README.txt:1]: should this be a md file?

File: org.eclipse.xtext.maven.plugin/pom.xml [hunk 2]
```diff
@@ -110,6 +115,21 @@
 			<version>${project.version}</version>
 			<scope>test</scope>
 		</dependency>
+		<!-- Needed by some of our IT projects:
```
Comment (cdietrich) [inline org.eclipse.xtext.maven.plugin/pom.xml:118]: wonder why these are not accessible if just mention in the tests pom

Comment (cdietrich) [reply org.eclipse.xtext.maven.plugin/pom.xml:118]: is this cause of upstream repos not availble in mono build

File: org.eclipse.xtext.maven.plugin/README.txt (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.maven.plugin/README.txt:1]: Yes, I'll transform that

File: org.eclipse.xtext.maven.plugin/pom.xml (hunk 2 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.maven.plugin/pom.xml:118]: My intention and understanding of integration tests for maven plugins are that our artifacts should NOT be downloaded from remote repositories; this way, we're sure we're testing the integration projects with our current artifacts (the ones built from the current sources). Now that we're on a single project, this makes sense.

That's why we use the invoker plugin to create a self-contained .m2 repo with the needed Xtext artifacts.
Note that in the IT projects I've disabled completely sonatype snapshots, I only left lsp4j taken from JIRO.
This way, we're sure that we are self-contained concerning our artifacts.

In general, despite split or mono repo, we made a terrible mistake in not using another .m2 repository for running integration tests, risking of corrupting the current .m2 repo.

Does that answer your question? Or did you mean something else?

File: org.eclipse.xtext.maven.plugin/README.txt (hunk 1 above)
Comment (LorenzoBettini) [reply org.eclipse.xtext.maven.plugin/README.txt]: Fixed in https://github.com/xtext/xtext-monorepo/pull/17/commits/4cd80eed85a3d758d9323edb2ec7171b6774f8d8

**Review by cdietrich — APPROVED** [review]

**Pull request merged**